ipatests/test_integration/base.py

```diff
@@ -97,9 +98,17 @@ def teardown_class(cls):
 
     @classmethod
     def uninstall(cls, mh):
-        tasks.uninstall_master(cls.master)
         for replica in cls.replicas:
+            try:
+                tasks.run_server_del(
+                    cls.master, replica.hostname, force=True,
+                    ignore_topology_disconnect=True, ignore_last_of_role=True)
+            except subprocess.CalledProcessError:
+                # If the master has already been uninstalled,
+                # this call may fail
+                pass
```

Comment:
@flo-renaud, hi.
Is it possible to narrow down an exception range to the re-uninstallation error only?

Reply:
Hi @stanislavlevin 
`tasks.run_server_del` will try to execute `kinit admin` on the master. If the master was already uninstalled, kinit fails and returns CalledProcessError.

Reply:
Since the current implementation of `run_server_del` just tries to perform `ipa server-del`, would it be better to pass `raiseonerr=False` down to `kinit_admin` too?

For example,

```diff
ipatests/pytest_ipa/integration/tasks.py:
 def run_server_del(host, server_to_delete, force=False,
                    ignore_topology_disconnect=False,
                    ignore_last_of_role=False):
-    kinit_admin(host)
+    kinit_admin(host, raiseonerr=False)
     args = ['ipa', 'server-del', server_to_delete]
     if force:

ipatests/test_integration/base.py:
-            try:
-                tasks.run_server_del(
-                    cls.master, replica.hostname, force=True,
-                    ignore_topology_disconnect=True, ignore_last_of_role=True)
-            except subprocess.CalledProcessError:
-                # If the master has already been uninstalled,
-                # this call may fail
-                pass
+            tasks.run_server_del(
+                cls.master, replica.hostname, force=True,
+                ignore_topology_disconnect=True, ignore_last_of_role=True)
```

Reply:
Hi @stanislavlevin 
with your suggestion, the behavior of run_server_del is modified when called on an uninstalled node: currently it's raising a CalledProcessError exception, with the change it would return the result code 1. That's why I preferred to handle the exception in the function base.py:uninstall.